ci/jenkins/stop-jobs.sh

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 Antrea Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eo pipefail
+
+function echoerr {
+    >&2 echo "$@"
+}
+
+PULL_REQUEST_ID=""
+JENKINS_URL=""
+DEFAULT_WORKDIR=$HOME
+
+_usage="Usage: $0 [--pull-request] [--jenkins]
+
+        --pull-request           Unique Pull Request id, to verify the stale jobs that needs to be aborted.
+        --jenkins                Jenkins url to fetch the information about jobs."
+
+function print_usage {
+    echoerr "$_usage"
+}
+
+function print_help {
+    echoerr "Try '$0 --help' for more information."
+}
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+    --pull-request)
+    PULL_REQUEST_ID="$2"
+    shift 2
+    ;;
+    --jenkins)
+    JENKINS_URL="$2"
+    shift 2
+    ;;
+    -h|--help)
+    print_usage
+    exit 0
+    ;;
+    *) 
+    echoerr "Unknown option $1"
+    exit 1
+    ;;
+esac
+done
+
+function abort_queue_jobs {
+    echo "====== checking for capv stale waiting jobs on PR ${PULL_REQUEST_ID} ======"
+    capv_job_name=$1
+    length_of_queue=$(curl -s -XGET "${JENKINS_URL}queue/api/json" | jq '[.[]][2]' | jq length)
```

Comment:
Is it better to save the `curl -s -XGET "${JENKINS_URL}queue/api/json"` output in an variable, and then when later it needs to parse it, just echo the variable and then pipe to jq. For example:

```
queue="$(curl -s -XGET "${JENKINS_URL}queue/api/json" )"
length_of_queue="$(echo "$queue" |  jq '[.[]][2]' | jq length)"

    for (( j=0; j<$length_of_queue; j++ )); do
        job_name="$(echo "$queue" | jq "[.[]][2][$j].task.name" | tr -d '"') "
        ...
    done
```

This saves some API calls to Jenkins.

Another concern is that, in the for loop line 69-82, after it aborts a job, I suppose that the curl command in line 70 will return a shorter list without the aborted job. Then the indices of the newly returned list won't match j because j is for the original list.

For example:
```
Jobs in queue: a, b, c, d, e, f.
Jobs to be aborted: b, c.
When j=1, it removes b, then j++, now j=2
Now jobs in queue: a, c, d, e,f
queue[j] is queue[2] is actually d, so c is skipped and not being checked by this loop.
```

If it doesn't fetch the latest queue, instead, it echo the saved queue json in a variable, there should be no such issue.